administrator/components/com_content/controllers/article.php

```diff
@@ -155,30 +155,47 @@ public function batch($model = null)
 	 *
 	 * @since	3.1
 	 */
-	protected function postSaveHook(JModelLegacy $model, $validData = array())
+	protected function postSaveHook(JModelLegacy $model, $validData)
 	{
 		JModelLegacy::addIncludePath(JPATH_ADMINISTRATOR . '/components/com_menus/models', 'MenusModel');
-		$itemmodel = JModelAdmin::getInstance('Item', 'MenusModel', array());
+		$itemModel = JModelAdmin::getInstance('Item', 'MenusModel', array());
+		$itemModel->addTablePath(JPATH_ADMINISTRATOR . '/components/com_menus/tables');
+
+		$db = JFactory::getDbo();
+		$query = $db->getQuery(true);
+		$query->select('extension_id');
+		$query->from($db->quoteName('#__extensions'));
+		$query->where($db->quoteName('name') . " = " . $db->quote('com_content'));
```

Comment:
Indeed. `JComponentHelper::getComponent('com_content')->id` is the best approach.